dev/com.ibm.ws.logging/src/com/ibm/ws/logging/internal/impl/ConsoleLogHandler.java

```diff
@@ -109,12 +109,17 @@ public void synchronousWrite(Object event) {
             //First retrieve a cached JSON  message if possible, if not, format it and store it.
             if (genData.getJsonMessage() == null) {
                 //check if it's in JSON format
-                String json = null;
-                if (event instanceof LogTraceData) {
-                    json = ((LogTraceData) event).getMessage();
-                }
-                if (isJSON(json) && appsWriteJson) {
-                    genData.setJsonMessage(json);
+
+                if (appsWriteJson) {
+                    String json = null;
+                    if (event instanceof LogTraceData) {
+                        json = ((LogTraceData) event).getMessage();
+                    }
+                    if (isJSON(json)) {
+                        genData.setJsonMessage(json);
+                    } else {
+                        genData.setJsonMessage((String) formatEvent(eventSourceName, CollectorConstants.MEMORY, event, null, MAXFIELDLENGTH));
+                    }
```

Comment:
I see your refactor and raise you a ...

                String jsonMessage = null;
                if (appsWriteJson && event instanceof LogTraceData) {
                    jsonMessage = ((LogTraceData) event).getMessage();
                    if (!isJSON(jsonMessage)) {
                        jsonMessage = (String)formatEvent(eventSourceName, CollectorConstants.MEMORY, event, null, MAXFIELDLENGTH);
                    }
                }

                genData.setJsonMessage(jsonMessage);
                
:-)